cg/apps/demultiplex/demultiplex_api.py

```diff
@@ -107,11 +112,20 @@ def get_stdout_logfile(flow_cell: FlowCellDirectoryData) -> Path:
         return Path(flow_cell.path, f"{DemultiplexingAPI.get_run_name(flow_cell)}.stdout")
 
     def flow_cell_out_dir_path(self, flow_cell: FlowCellDirectoryData) -> Path:
-        """Create the path to where the demuliplexed result should be produced."""
+        """Create the path to where the demultiplexed result should be produced."""
         return Path(self.demultiplexed_runs_dir, flow_cell.path.name)
 
-    def unaligned_dir_path(self, flow_cell: FlowCellDirectoryData) -> Path:
-        """Create the path to where the demuliplexed result should be produced."""
+    def sample_sheet_exists_in_hk(self, flow_cell: FlowCellDirectoryData) -> bool:
```

Comment:
```suggestion
    def get_sample_sheet(self, flow_cell_id: str) -> Optional[File]:
```
I think the function actually does this and you can send in as little as possible to the function. Only what you need and not the whole object

Reply:
if you need a existent check make a separate bool function to not have side effects

Reply:
Fixed